features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -1123,6 +1125,53 @@ static nsapi_error_t mbed_lwip_setsockopt(nsapi_stack_t *stack, nsapi_socket_t h
             }
             return 0;
 
+        case NSAPI_ADD_MEMBERSHIP:
+        case NSAPI_DROP_MEMBERSHIP: {
+            if (optlen != sizeof(nsapi_ip_mreq_t)) {
+                return NSAPI_ERROR_PARAMETER;
+            }
+            err_t igmp_err;
+            const nsapi_ip_mreq_t *imr = (const nsapi_ip_mreq_t*) optval;
```

Comment:
Unnecessary C-style cast is unhelpful - would have hidden the const-correctness error previously reported.